### PR TITLE
Social Previews: Add the ability to preview Twitter Threads

### PR DESCRIPTION
**Pull request by pento**
#### Changes proposed in this Pull Request

This PR adds the ability to preview entire Twitter threads.

It also makes the single Tweet preview appear more like a tweet, adding support for Twitter account details to be passed, as well as Publicize's custom sharing message.

The thread portion of this PR is being built in conjunction with Automattic/jetpack#16699.

##### WordPress.com Calypso Editor

To set your test site to the Calypso Editor, visit your user RC, find the site in the _Selected Editors_ settings, and clear the setting for web.

Functionality can be tested at calypso.live/?branch=add/tweetstorm-social-preview. Aside from the additional bits of Twitter preview UI, functionality should remain the same.

##### WordPress.com Block Editor

This change will be shipped to the block editor through Jetpack, there currently isn't a diff setup for testing.

##### In Jetpack master

- Switch your Jetpack branch to `master`.
- In your `wp-calypso` directory, `cd` into `packages/social-previews`, then run `yarn link`.
- In your `jetpack` directory, run `yarn link "@automattic/social-previews"`, then `yarn build-extensions`.

Functionality can then be tested in your Jetpack dev environment. Aside from the additional bits of Twitter preview UI, functionality should remain the same.

##### In Jetpack PR Automattic/jetpack#16699

- Switch your Jetpack branch to ` add/publishing-tweetstorms`.
- If you haven't already, run the two `yarn link` commands above, then build Jetpack.

Functionality can then be tested in your Jetpack dev environment. Default behaviour should stay the same, thread mode can be tested by selecting the "Share your entire blog post as a Twitter thread in multiple Tweets" option in Jetpack's sharing panel.

### CODE REVIEW

File: packages/social-previews/src/twitter-preview/style.scss [hunk 1]
```diff
@@ -1,112 +1,223 @@
...
- * such as Google and Facebook. Therefore there
+ * designed to match the CSS on Twitter. Therefore there
  * will be exceptions to our CSS guidelines here
- * but please do not "update" this file to
- * conform.
+ * but please do not "update" this file to conform.
  *
- * @blame: dmsnell
-*/
+ * @blame: pento
```
Comment (obenland) [inline packages/social-previews/src/twitter-preview/style.scss:7]: 👍

File: packages/social-previews/src/twitter-preview/style.scss [hunk 2]
```diff
@@ -1,112 +1,225 @@
...
+				}
+			}
+
+			&.twitter-preview__media-children-2 {
+				grid-template-areas: 'a b';
+			}
+
+			&.twitter-preview__media-children-3 {
+				grid-template-areas: 'a b'
+									 'a c';
```
Comment (marekhrabe) [inline packages/social-previews/src/twitter-preview/style.scss:107]: Is this formatting right? It looks really strange here on Github

**Review by marekhrabe — COMMENTED** [review]
I've noted some minor things, otherwise this is working well for me in both the context of Calypso and Jetpack.

File: packages/social-previews/src/twitter-preview/style.scss (hunk 2 above)
Comment (pento) [reply packages/social-previews/src/twitter-preview/style.scss]: The alignment indenting got converted to tabs at some point. 🙂 Fixed in c2af41cd0a.

File: packages/social-previews/src/twitter-preview/index.jsx [hunk 3]
```diff
@@ -1,64 +1,59 @@
...
+				text: '',
+				media: [],
+				card: {
+					...this.props,
+					type: 'large_image_summary' === this.props.type ? 'summary_large_image' : this.props.type,
+				},
+				date: Date.now(),
+				name: 'Account Name',
+				profileImage:
+					'https://abs.twimg.com/sticky/default_profile_images/default_profile_bigger.png',
```
[the remark this answers is not in the document]
Comment (pento) [reply packages/social-previews/src/twitter-preview/index.jsx:33]: I suspect that the bigger risk is in us hosting the image, then Twitter changes the default image. This is the same default image URL that Twitter has always used, even when it was the old "egg" image.

The URL (and its variants) are [documented](https://developer.twitter.com/en/docs/twitter-api/v1/accounts-and-users/user-profile-images-and-banners), if Twitter were to ever change it, there would be a lot of apps that break: I suspect they would be as reluctant to change that URL as we would be.

**Comment by pento** [comment]
Thank you for the review, @marekhrabe! I've addressed your feedback, let me know if anything else jumps out at you. 🙂

**Review by marekhrabe — APPROVED** [review]

**Pull request merged**